var/spack/repos/builtin/packages/r-dt/package.py

```diff
@@ -16,8 +16,13 @@ class RDt(RPackage):
     url      = "https://cran.r-project.org/src/contrib/DT_0.1.tar.gz"
     list_url = "https://cran.r-project.org/src/contrib/Archive/DT"
 
-    version('0.1', '5c8df984921fa484784ec4b8a4fb6f3c')
+    version('0.5', sha256='0e5bbb91b88a769e52ba079ebac266a25ea0f0a23bdf3deff31f51d2a81eade8')
+    version('0.4', sha256='3daa96b819ca54e5fbc2c7d78cb3637982a2d44be58cea0683663b71cfc7fa19')
+    version('0.3', sha256='ef42b24c9ea6cfa1ce089687bf858d773ac495dc80756d4475234e979bd437eb')
+    version('0.2', sha256='a1b7f9e5c31a241fdf78ac582499f346e915ff948554980bbc2262c924b806bd')
+    version('0.1', sha256='129bdafededbdcc3279d63b16f00c885b215f23cab2edfe33c9cbe177c8c4756')
 
-    depends_on('r-htmltools', type=('build', 'run'))
-    depends_on('r-htmlwidgets', type=('build', 'run'))
+    depends_on('r-htmltools@0.3.6:', type=('build', 'run'))
```

Comment:
Do these tighter restrictions apply to *all* versions of `dt`, or just recent releases?

Reply:
I'm usually lazy and allow these kinds of tighter restrictions unless someone really does need an older version of the dependency. This is a common problem for Python and R packages, and I don't like to make people search backwards through every version to see when the dependency version was upgraded. But you are correct, it is more technically correct to list which version of `r-dt` require which versions of `r-htmltools`. If someone needs `r-dt` to build with an older version of `r-htmltools`, they are free to hack this package and submit a PR. In the meantime, I'm going to merge.